ocs_ci/utility/networking.py

```
@@ -244,41 +244,60 @@ def add_data_replication_separation_to_cluster_data(cluster_data):
+
+    # Get private IP and prefix from the first worker node using ip addr command
+    interface, private_ip, prefix_length = get_node_private_ip(worker_nodes[0])
 
+    ip_network = ipaddress.IPv4Network(
+        f"{private_ip}/{prefix_length}",
+        strict=False,
+    )
+    str_network = f"{ip_network.network_address}/{ip_network.prefixlen}"
+
+    if "spec" not in cluster_data or not isinstance(cluster_data["spec"], dict):
+        raise ValueError("cluster_data missing or invalid 'spec' field")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)